### DISCUSSION TITLE
useQuery inside modal

### DISCUSSION THREAD

**Discussion by cksal0805** [ant-design/ant-design-mobile#5297] [Q&A] [answered]
I have a modal that gets attached inside my <div id="root" /> which contains my app (wrapped in QueryClientProvider). But I get the following error when that modal gets mounted/attached:

No QueryClient set, use QueryClientProvider to set one
I've got other useQuery() hooks working throughout the app but not in this modal. What am I missing here?

**Comment by awmleer** [comment, marked as answer]
If you use Modal.show, the content inside it are rendered in another React app, which is created by antd-mobile. So the context here would be isolated.

You can directly render `<Modal />` in your component and this problem won't exist.
